packages/eslint-plugin/src/rules/no-useless-empty-export.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+function isEmptyExport(
+  node: TSESTree.Node,
+): node is TSESTree.ExportNamedDeclaration {
+  return (
+    node.type === AST_NODE_TYPES.ExportNamedDeclaration &&
+    node.specifiers.length === 0 &&
+    !node.declaration
+  );
+}
+
+const exportOrImportNodeTypes = new Set([
+  AST_NODE_TYPES.ExportAllDeclaration,
+  AST_NODE_TYPES.ExportDefaultDeclaration,
+  AST_NODE_TYPES.ExportNamedDeclaration,
+  AST_NODE_TYPES.ExportSpecifier,
+  AST_NODE_TYPES.ImportDeclaration,
+  AST_NODE_TYPES.TSExportAssignment,
+  AST_NODE_TYPES.TSImportEqualsDeclaration,
+]);
+
+export default util.createRule({
+  name: 'no-useless-empty-export',
+  meta: {
+    docs: {
+      description:
+        "Disallow empty exports that don't change anything in a module file",
+      recommended: false,
+      suggestion: true,
+    },
+    fixable: 'code',
+    hasSuggestions: true,
+    messages: {
+      uselessExport: 'Empty export does nothing and can be removed.',
+    },
+    schema: [],
+    type: 'suggestion',
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      Program(node): void {
```

Comment:
nit - it'd be good if this handled `declare module`/`declare namespace`:
https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEBbA9sArhBAzZz4G8BYAKHjPhAA8AHZGAF3jGQDsBnRy+AXngEYA3CRLkKNOo3wBfIcSlA

note that without a `declare`, an empty `export {}` is semantically illegal!

Reply:
> note that without a declare, an empty export {} is semantically illegal!

Sorry, I don't follow - is that a change request?

Reply:
No that's just an interesting titbit about the language that I learned double checking the syntax!